docker-compose.yml

```diff
@@ -0,0 +1,174 @@
+    ]
+    expose:
+      - 9092
+      - 29092
+    depends_on:
+      - zookeeper
+    environment:
+      KAFKA_AUTO_CREATE_TOPICS_ENABLE: true
+      LOG_DIR: "/tmp/logs"
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_INTER_BROKER_LISTENER_NAME: DOCKER
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: DOCKER:PLAINTEXT,LOCALHOST:PLAINTEXT
+      KAFKA_LISTENERS: DOCKER://kafka:29092,LOCALHOST://localhost:9092
+      KAFKA_ADVERTISED_LISTENERS: DOCKER://kafka:29092,LOCALHOST://localhost:9092
+    volumes:
+      - ./compose-data/kafka:/tmp/logs
+
+  init-kafka:
+    profiles:
+      - "kafka"
+    image: quay.io/strimzi/kafka:latest-kafka-3.4.0
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --list
+
+      echo -e 'Creating kafka topics'
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.sources.status --replication-factor 1 --partitions 1
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.provisioning.internal.availability-check --replication-factor 1 --partitions 1
+
+      echo -e 'Successfully created the following topics:'
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --list
+      "
+    depends_on:
+      - kafka
+      - zookeeper
+  
+  backend:
+    env_file:
+      - scripts/compose/api.compose.env
+    depends_on:
+      migrate:
+        condition: service_completed_successfully
+      provisioning-db:
+        condition: service_healthy
+      init-kafka:
+        condition: service_completed_successfully
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    ports:
+      - ${APP_PORT:-8000}:${APP_PORT:-8000}
+      - ${PROMETHEUS_PORT:-9000}:${PROMETHEUS_PORT:-9000}
+    volumes:
+      - ./:/backend
+
+  migrate: # This service is used to run migrations
+    profiles:
+      - "migrate"
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    env_file:
+      - config/api.env
+    depends_on:
+      provisioning-db:
+        condition: service_healthy
+    entrypoint: [go,run,/build/cmd/pbackend migrate]
+
+  frontend:
+    profiles:
+      - "frontend"
+    environment:
+      - PROV_API_HOST=backend
+      - WATCHPACK_POLLING=true
+    build:
+      context: ../provisioning-frontend
+      dockerfile: Dockerfile.dev
+    volumes:
+      - ../provisioning-frontend/:/prov-frontend
+    ports: 
+      - 1337:1337
+    depends_on:
+    - backend
+
+  sources:
+    profiles:
+      - "sources"
+    build:
+      context: ../sources-api-go
+    ports: 
+      - 8131:8131
+    env_file:
+      - scripts/compose/sources.compose.env
+    depends_on:
+      redis:
+        condition: service_healthy
+      sources-db:
+        condition: service_healthy
```

Comment:
I would also suggest using git submodules for sources (paths need an update). Can you introduce new profile named "sources-dev" and rename this service tag to that? Then also add this:

```yaml
 sources:
    profiles:
      - "sources"
    container_name: sources
    image: quay.io/cloudservices/sources-api-go:latest
    env_file:
      - sources.env
    expose:
      - 8131
    depends_on:
      redis:
        condition: service_healthy
      sources-db:
        condition: service_healthy
```

After you do `podman login quay.io` this will work so people can simply use the official sources. You can do the same for notifications or other containers as well. Users can decide the profile they want to use any time. It can share the same database that is actually beneficial.